tests/azure_iot_ut.c

```diff
@@ -0,0 +1,172 @@
+                                                            ucTestValue, sizeof( ucTestValue ) - 1 ),
+                          eAzureIoTSuccess );      
+}
+/*-----------------------------------------------------------*/
+
+static void testAzureIoTMessagePropertiesAppend_Success( void ** ppvState )
+{
+    AzureIoTMessageProperties_t xTestMessageProperties;
+
+    ( void ) ppvState;
+
+    assert_int_equal( AzureIoT_MessagePropertiesInit( &xTestMessageProperties,
+                                                      ucBuffer, 0, sizeof( ucBuffer ) ),
+                      eAzureIoTSuccess );
+
+    assert_int_equal( AzureIoT_MessagePropertiesAppend( &xTestMessageProperties,
+                                                        ucTestKey, sizeof( ucTestKey ) - 1,
+                                                        ucTestValue, sizeof( ucTestValue ) - 1 ),
+                      eAzureIoTSuccess );   
+}
+/*-----------------------------------------------------------*/
+
+static void testAzureIoTMessagePropertiesFind_Failure( void ** ppvState )
+{
+    AzureIoTMessageProperties_t xTestMessageProperties;
+    const uint8_t * pucOutValue;
+    uint32_t ulOutValueLength;
+
+    ( void ) ppvState;
+
+    assert_int_equal( AzureIoT_MessagePropertiesInit( &xTestMessageProperties,
+                                                      ucBuffer, 0, sizeof( ucBuffer ) ),
+                      eAzureIoTSuccess );
+
+    assert_int_equal( AzureIoT_MessagePropertiesAppend( &xTestMessageProperties,
+                                                        ucTestKey, sizeof( ucTestKey ) - 1,
+                                                        ucTestValue, sizeof( ucTestValue ) - 1 ),
+                      eAzureIoTSuccess );
+
+    /* Failed for NULL key */
+    assert_int_not_equal( AzureIoT_MessagePropertiesFind( &xTestMessageProperties,
+                                                          NULL, 0, &pucOutValue, &ulOutValueLength ),
+                          eAzureIoTSuccess );
+
+    /* Failed for NULL outvalue */
+    assert_int_not_equal( AzureIoT_MessagePropertiesFind( &xTestMessageProperties,
+                                                          ucTestKey, sizeof( ucTestKey ) - 1,
+                                                          NULL, 0 ),
+                          eAzureIoTSuccess ); 
+}
+/*-----------------------------------------------------------*/
+
+static void testAzureIoTMessagePropertiesFind_Success( void ** ppvState )
+{
+    AzureIoTMessageProperties_t xTestMessageProperties;
+    const uint8_t * pucOutValue;
+    uint32_t ulOutValueLength;
+
+    ( void ) ppvState;
+
+    assert_int_equal( AzureIoT_MessagePropertiesInit( &xTestMessageProperties,
+                                                      ucBuffer, 0, sizeof( ucBuffer ) ),
+                      eAzureIoTSuccess );
+
+    assert_int_equal( AzureIoT_MessagePropertiesAppend( &xTestMessageProperties,
```

Comment:
Even though it's technically re-testing the embedded SDK and I don't want to over do that, any chance you could do a few more append() here and search for those?  I typically find 3 to be a reasonable number on a passthrough like this.